libchisel/src/lib.rs

```diff
@@ -54,6 +59,51 @@ pub trait ChiselModule<'a> {
     fn as_abstract(&'a self) -> Self::ObjectReference;
 }
 
+pub trait ModuleConfig {
+    type ConfigKey: Clone + Hash + Eq;
+    type ConfigValue: Clone;
+    type Configuration: KeyValue<Self::ConfigKey, Self::ConfigValue> =
+        HashMap<Self::ConfigKey, Self::ConfigValue>;
+
+    fn with_config(config: &Self::Configuration) -> Self;
+}
+
+pub trait KeyValue<K, V>
+where
+    K: Clone + Hash + Eq,
+    V: Clone,
+{
+    fn insert(&mut self, key: &K, value: &V) -> Option<V>;
+
+    fn get(&self, key: &K) -> Option<&V>;
+
+    fn get_mut(&mut self, key: &K) -> Option<&mut V>;
+
+    fn remove(&mut self, key: &K) -> Option<V>;
```

Comment:
Only `get` is needed for chisel.

Reply:
Yeah, I know. I was treating this as a general abstraction for keyvalue containers.